packages/editor/src/components/dialogs/PublishModal.tsx

```diff
@@ -0,0 +1,271 @@
+    }
+
+    try {
+      if (location?.id) {
+        await locationMutation.patch(location.id, locationData)
+      } else {
+        await locationMutation.create(locationData)
+      }
+      PopoverState.hidePopupover()
+    } catch (err) {
+      errors.serverError.set(err.message)
+    }
+    submitLoading.set(false)
+  }
+
+  const unpublishLocation = async () => {
+    if (location?.id) {
+      submitLoading.set(true)
+      try {
+        await locationMutation.remove(location.id)
+        PopoverState.hidePopupover()
+      } catch (err) {
+        errors.serverError.set(err.message)
+        submitLoading.set(false)
+      }
+    }
+  }
+
+  return (
+    <div className="relative z-50 max-h-[80vh] w-[50vw] bg-theme-surface-main">
+      <div className="relative rounded-lg shadow">
+        <ModalHeader
+          onClose={PopoverState.hidePopupover}
+          title={location?.id ? t('editor:toolbar.publishLocation.create') : t('editor:toolbar.publishLocation.update')}
+        />
+        <div className="h-fit max-h-[60vh] w-full overflow-y-auto px-10 py-6">
+          <div className="relative grid w-full gap-6">
+            {errors.serverError.value && <p className="mb-3 text-red-700">{errors.serverError.value}</p>}
+            {location && (
+              <Button
+                size="medium"
+                variant="transparent"
+                className="w-full cursor-default text-left text-xs"
+                endIcon={
+                  <HiLink
+                    className="z-10 h-4 w-4 cursor-pointer"
+                    onClick={() => {
+                      navigator.clipboard.writeText(
+                        new URL(window.location.origin + `/location/${location.slugifiedName}`).href
+                      )
+                      NotificationService.dispatchNotify(t('editor:toolbar.publishLocation.locationLinkCopied'), {
+                        variant: 'success'
+                      })
+                    }}
+                  />
+                }
+              >
+                <div
+                  className="cursor-pointer text-blue-primary hover:underline"
+                  onClick={() => window.open(new URL(location.url))}
+                >
+                  {location.url}
+                </div>
+              </Button>
+            )}
+            <Input
+              label={t('admin:components.location.lbl-name')}
+              value={name.value}
+              onChange={(event) => name.set(event.target.value)}
+              error={errors.name.value}
+              disabled={submitLoading.value}
+            />
+            <Input
+              type="number"
+              label={t('admin:components.location.lbl-maxuser')}
+              value={maxUsers.value}
+              onChange={(event) => maxUsers.set(Math.max(parseInt(event.target.value, 0), 0))}
+              error={errors.maxUsers.value}
+              disabled={submitLoading.value}
+            />
+            <Select
```

Comment:
The values shown in this dropdown are wrong.

Also why do we need to show this dropdown at all? Because we are essentially in a scene publishing it. Maybe publish button appears only when we have a scene selected?

Reply:
disabled it for studio publish button.
yes, the publish button is only visible when a scene is selected